frontend/e2e/pages/metrics/metrics-step.ts

```diff
@@ -779,4 +808,40 @@ export class MetricsStep {
       'Please go back to the previous page and change your pipeline token with correct access permission.',
     );
   }
+
+  async checkPartialApiFailedTimeRangeIndicator(rangeCount: number) {
+    await this.dateRangeViewerExpandTrigger.click();
+    await expect(this.dateRangeViewerOptions).toBeVisible();
+    const apiIndicators = {
+      green: 0,
+      red: 0,
+    };
+    for (let i = 0; i < rangeCount; i++) {
+      const currentRange = this.dateRangeViewerOptions.getByLabel(`date range viewer - option ${i}`);
+      if (currentRange.getByTestId('PriorityHighIcon')) {
+        apiIndicators.red++;
+      }
+      // todo green++
+      // if (currentRange.getByTestId('PriorityHighIcon')) {
+      //   apiIndicators.red++
+      // }
+    }
+    expect(apiIndicators.red).toBeGreaterThan(0);
+    // expect(apiIndicators.green).toBeGreaterThan(0)
+  }
+
+  async checkApiFailedAlertVisible() {
+    await expect(this.page.getByRole('alert').first()).toContainText(
+      'Failed to get partial Board configuration, you can click "Next" button to go to Report page.',
+    );
+  }
+
+  async checkSomeApiFailed(rangeCount: number) {
+    await this.checkApiFailedAlertVisible();
+    await this.checkPartialApiFailedTimeRangeIndicator(rangeCount);
+    // await expect(this.page.getByRole('alert').first()).toContainText('Help Information');
+    // await expect(this.page.getByRole('alert').first()).toContainText(
+    //   'The file will expire in 30 minutes, please download it in time.',
+    // );
```

Comment:
???

Reply:
waiting for another PR to merge into, don't worry.

Reply:
done